src/app/modules/dashboard/components/course-list/course-list.component.ts

```diff
@@ -48,11 +57,14 @@ export class CourseListComponent implements OnInit, OnChanges {
   public deleteCourse(id: string) {
     let response = this.displayModal()
     if (response) {
-      this.courseService.removeCourse(id)
-        .then(() => {
-          this.courseService.getList()
-            .then((courses: ICourse[]) => {
-              this.courses = courses;
+      this._courseService.removeCourse(id)
+        .subscribe(() => {
+          this._courseService.getList()
+            .subscribe((courses: ICourse[]) => {
+              setTimeout(() => {
```

Comment:
why we are using setTimeout here?

Reply:
I wanted to mock data loading process in a more demonstrative way :)